extensions/amp-inputmask/0.1/amp-inputmask.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {TextMask} from './text-mask';
+import {isExperimentOn} from '../../../src/experiments';
+import {user} from '../../../src/log';
+
+const SERVICE = 'inputmask';
+const TAG = `amp-${SERVICE}`;
+
+export class AmpInputmaskService {
+  /**
+   * @param {!../../../src/service/ampdoc-impl.AmpDoc} ampdoc
+   */
+  constructor(ampdoc) {
+    this.ampdoc = ampdoc;
+
+    /** @private {!Array<!TextMask>} */
+    this.masks_ = [];
+  }
+
+  /**
+   * Install the inputmask service and controllers.
+   */
+  install() {
+    const maskElements = this.ampdoc.getRootNode().querySelectorAll('[mask]');
```

Comment:
we would need to redo this for when amp-list adds new inputs. So need something like
```
   doc.addEventListener(AmpEvents.DOM_UPDATE, () => {
      // query and install again
    });
```